### DISCUSSION TITLE
[Feature Request] Add tags/substrings to excludes some informations in report

### DISCUSSION THREAD

**Discussion by TriiniitVB** [DannyDainton/newman-reporter-htmlextra#402] [Ideas]
Hi, 

I use the newman-reporter-htmlextra with Azure DevOps pipeline, but there are some informations I want to hide.

It would be great to have an option like "hideRequestBody" for hide request header and  request body only with a tag or substrings, who check the name or/and the description on the request, and hide or not the information.

**For exemple**
i've 3 requests with the following names
1 - Create Account - SENSITIVE
2 - Get informations about referential
3 - [SENSITIVE] - Log In account

**How** --> `--reporter-htmlextra-hidebytag "SENSITIVE" `
**Result** --> In the final report we don't see the request header/body for requests number 1 and 3. But we have all the informations about request number 2

Thanks

**Comment by w4dd325** [comment]
Hi @TriiniitVB 

There is the option to create customised template files that you can use to output your results in specific ways. 
You can utilise them using the CLI option;  --reporter-htmlextra-template <path>

There are also some other CLI options that allow you to hide data, for example;

--reporter-htmlextra-omitResponseBodies
An optional flag which allows you to exclude all Response Bodies from the final report

--reporter-htmlextra-hideRequestBody
An optional flag which allows you to exclude certain Request Bodies from the final report. Enter the name of the request that you wish to hide.

--reporter-htmlextra-hideResponseBody
An optional flag which allows you to exclude certain Response Bodies from the final report. Enter the name of the request that you wish to hide.

If you are interested in adding additional functionality, this is an open-source project, I recently contributed (my first public PR) to this project as we use it all the time! I would highly recommend getting involved!

There is a section called "Contributing To The Project" that you can find here;
https://www.npmjs.com/package/newman-reporter-htmlextra?activeTab=readme
